From 2427780fdab86e70e4e3152c3aa40b161fe9c1a4 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Thu, 03 Mar 2022 09:35:49 +0000
Subject: [PATCH] IWYU: missing include for usage of std::unique_ptr in vulkan_function_pointers.h

Added <memory> include.

Fixes build error with libstdc++:
    ./../gpu/vulkan/vulkan_function_pointers.h:77:32: error: ‘unique_ptr’ is not a member of ‘std’
       77 |   base::flat_map<VkQueue, std::unique_ptr<base::Lock>> per_queue_lock_map;
          |                                ^~~~~~~~~~

Bug: 957519
Change-Id: Iefdf89112d5f441071135731cdbc057ff7ec0f3a
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3499100
Reviewed-by: Peng Huang <penghuang@chromium.org>
Commit-Queue: José Dapena Paz <jdapena@igalia.com>
Cr-Commit-Position: refs/heads/main@{#977028}
---

diff --git a/gpu/vulkan/generate_bindings.py b/gpu/vulkan/generate_bindings.py
index d9ca37e..6b506da2 100755
--- a/gpu/vulkan/generate_bindings.py
+++ b/gpu/vulkan/generate_bindings.py
@@ -389,6 +389,8 @@
 
 #include <vulkan/vulkan.h>
 
+#include <memory>
+
 #include "base/compiler_specific.h"
 #include "base/component_export.h"
 #include "base/containers/flat_map.h"
diff --git a/gpu/vulkan/vulkan_function_pointers.h b/gpu/vulkan/vulkan_function_pointers.h
index 8a824a2..f6b0556 100644
--- a/gpu/vulkan/vulkan_function_pointers.h
+++ b/gpu/vulkan/vulkan_function_pointers.h
@@ -13,6 +13,8 @@
 
 #include <vulkan/vulkan.h>
 
+#include <memory>
+
 #include "base/compiler_specific.h"
 #include "base/component_export.h"
 #include "base/containers/flat_map.h"
